client.go

```diff
@@ -951,9 +970,8 @@ func (f *File) readChunkAt(ch chan result, b []byte, off int64) (n int, err erro
 // the number of bytes read and an error, if any. ReadAt follows io.ReaderAt semantics,
 // so the file offset is not altered during the read.
 func (f *File) ReadAt(b []byte, off int64) (int, error) {
-	if len(b) <= f.c.maxPacket {
-		// This should be able to be serviced with 1/2 requests.
-		// So, just do it directly.
+	if len(b) <= f.c.maxPacket || f.c.disableConcurrentReads {
+		// This should be able to be serviced with 1/2 requests or concurrent reads are disabled.
 		return f.readChunkAt(nil, b, off)
```

Comment:
I just realized, this is calling `readChunkAt` which is only intended to handle buffers up to `f.c.maxPacket` size. Its design has us request the whole length of the buffer passed in, which could feasibly be well above `f.c.maxPacket`.

We’re probably going to want a different receiver method that handles the `ReadAt` sequentially, but still breaks it up into `f.c.maxPacket` blocked requests to `readChunkAt`.

Maybe “chunk” as a name didn’t serve its purpose of making its functionality clear…